src/lib/BulkEmailChecker.js

```diff
@@ -0,0 +1,265 @@
+/* flow */
+/* eslint-disable no-console, no-param-reassign */
+
+import dns from 'dns';
+import { Socket } from 'net';
+import EventEmitter from 'events';
+import promisify from './promisify';
+import winston from 'winston';
+
+function createSmtpConnection(mx: string): Promise<any> {
+	const socket = new Socket();
+	const connection = socket.connect(25, mx);
+	let resolve, reject;
+
+	const send = command => {
+		connection.write(command);
+		connection.write('\r\n');
+		return new Promise((res, rej) => {
+			resolve = res;
+			reject = rej;
+		});
+	};
+
+	const close = () => {
+		connection.removeAllListeners();
+		socket.destroy();
+	};
+
+	function clear() {
+		resolve = null;
+		reject = null;
+	}
+
+	connection.setEncoding('ascii');
+
+	connection.on('connect', () => {
+		winston.info('EMAIL-SCRUBBER', `connection established for the mx: ${mx}`);
+	});
+
+	connection.on('data', data => {
+		try {
+			const [ , responseCode, msg ] = data.match(/\n?(\d{3})\s+(.*)/);
+			if (resolve) {
+				resolve({
+					responseCode: parseInt(responseCode, 10),
+					msg,
+					send,
+					close
```

Comment:
It is strange to return references to `send` and `close` in responses to `send()`

When response Code is 220, it is guaranteed that resolve belongs to the promise returned by `createSmtpConnection()`, so you can return all these properties ({ responseCode, message, send, close });

In other cases, just send responseCode and message.